<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.549">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Erika Carlson">
<meta name="dcterms.date" content="2024-03-06">

<title>data-analysis-replication</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="data-analysis-replication_files/libs/clipboard/clipboard.min.js"></script>
<script src="data-analysis-replication_files/libs/quarto-html/quarto.js"></script>
<script src="data-analysis-replication_files/libs/quarto-html/popper.min.js"></script>
<script src="data-analysis-replication_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="data-analysis-replication_files/libs/quarto-html/anchor.min.js"></script>
<link href="data-analysis-replication_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="data-analysis-replication_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="data-analysis-replication_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="data-analysis-replication_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="data-analysis-replication_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">data-analysis-replication</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Erika Carlson </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">March 6, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="data-analysis-replication" class="level1 unnumbered">
<h1 class="unnumbered">Data Analysis Replication</h1>
<p><strong>Minimum:</strong> 3 analyses, one descriptive statistical analysis, one visualization, and one inferential statistical analysis.</p>
<section id="introduction" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="introduction">Introduction</h2>
<p>Microglia are central nervous system (CNS)-resident immune cells with key roles in many neurological diseases. Microglia morphology changes in response to stimuli and is often used as a proxy for neuroinflammation and/or pathology. This paper tested for comparability across five common ImageJ (NIH) methods used to assess microglia morphology in mice. Analyses were conducted on identical photomicrographs, as well as individual microglia cells within the photomicrographs for direct comparison. The results of the five methods indicated a lack of comparability, leading the authors to conclude that studies should combine multiple methods to achieve an accurate representation of microglia morphology.</p>
<p>This paper conducted five ImageJ-based analysis techniques, two on full 40x photomicrographs and three on single microglia cells isolated from those photomicrographs. These photomicrographs were stained for Ionized calcium binding adapter molecule 1 (Iba1). Iba1 is a cannonical microglial marker used to examine microglial reactivity. The techniques were applied to identify alterations in microglial morphology or percent coverage between <strong>Treatment</strong> groups (Control vs PLX) that had pharmacologically manipulated microglia following an inflammatory stimuli. The five analysis techniques were:</p>
<ol type="1">
<li>Percent coverage of Iba1 staining to show how much Iba1 positive staining is present in a given photomicrograph.</li>
<li>Full photomicrograph skeletal analysis to calculate an averaged number of branches, branch endpoints, and branch length among cells in the field of view.</li>
<li>Fractal analysis to quantify the spatial complexity of the individually isolated microglia.</li>
<li>Single cell skeletal analysis to quantify microglial ramification and cell body size.</li>
<li>Sholl analysis to determine the extent of branching of individual microglia via intercepts on concentric circles around the cell body.</li>
</ol>
<p>Authors then fit hierarchical generalized linear mixed models using {glmmTMB} package in <strong><em>R</em></strong>. The error distribution (beta, negative-binomial, gaussian, gamma) used depended on the scale of the response variable. All models included a fixed effect for treatment. Sholl analysis included additional effects, described below. Random effects structure was used for aggregated, single cell, and Sholl analysis data sets. Inferences were based on a combination of coefficient estimates (β) and their 95% confidence intervals, differences between predicted conditional means (∆), effect sizes (d), and p-values following Tukey’s adjustments for multiple comparisons using {emmeans} package in <strong><em>R</em></strong>.</p>
<p>Here, I have replicated an aggregated analysis of Iba1 coverage, a single cell analysis of Iba1 coverage, and Sholl analysis of cell complexity. I also replicated the associated visualizations.</p>
</section>
<section id="preliminaries" class="level2">
<h2 class="anchored" data-anchor-id="preliminaries">Preliminaries</h2>
<p>Load required packages</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># data wrangling and ggplot2</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(skimr) <span class="co"># data wrangling</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggprism) <span class="co"># data visualization</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr) <span class="co"># data visualization</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("glmmTMB")</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmmTMB) <span class="co"># frequentist model fitting</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages("ggprism")</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans) <span class="co"># statistics</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Tell {knitr} where to output all of the figures associated with code chunks and where to find any images to include in the “.html” output.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">fig.path =</span> <span class="st">"images/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read in data files</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>perc_coverage <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/Perc_Coverage.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 225 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): Treatment
dbl  (2): Coverage, Perc_Cov
time (1): Animal_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(perc_coverage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 4
  Animal_ID Treatment Coverage Perc_Cov
  &lt;time&gt;    &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt;
1 01:01     Control       15.5    0.155
2 01:01     Control       18.3    0.183
3 01:01     Control       17.6    0.176
4 01:01     Control       12.9    0.129
5 01:01     Control       23.8    0.238
6 01:01     Control       17.6    0.176</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>perc_coverage_avg <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/Perc_Coverage_Avg.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 25 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): Treatment
dbl  (2): Avg_percent, Perc_Cov
time (1): Animal_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(perc_coverage_avg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 4
  Animal_ID Treatment Avg_percent Perc_Cov
  &lt;time&gt;    &lt;chr&gt;           &lt;dbl&gt;    &lt;dbl&gt;
1 01:01     Control          17.5    0.175
2 01:05     Control          16.7    0.167
3 01:06     Control          12.3    0.123
4 01:07     Control          17.4    0.174
5 01:08     Control          19.7    0.197
6 01:12     PLX              14.0    0.140</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>single_cell <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/Single_Cell.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 675 Columns: 16
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (1): Treatment
dbl  (14): Cohort, Region, FractalDimension(DB), Lacunarity, Density(pixels/...
time  (1): Animal_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(single_cell)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 16
  Animal_ID Cohort Treatment Region `FractalDimension(DB)` Lacunarity
  &lt;time&gt;     &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt;                  &lt;dbl&gt;      &lt;dbl&gt;
1 01:01          1 Control        2                   1.53      0.485
2 01:01          1 Control        2                   1.56      0.324
3 01:01          1 Control        1                   1.42      0.734
4 01:01          1 Control        2                   1.62      0.316
5 01:01          1 Control        3                   1.48      0.450
6 01:01          1 Control        2                   1.63      0.301
# ℹ 10 more variables: `Density(pixels/area)` &lt;dbl&gt;, SpanRatio &lt;dbl&gt;,
#   Circularity &lt;dbl&gt;, NumberOfEndpoints &lt;dbl&gt;, `BranchLength(um)` &lt;dbl&gt;,
#   NumberBranches &lt;dbl&gt;, body_area &lt;dbl&gt;, cell_area_C &lt;dbl&gt;,
#   body_perimeter &lt;dbl&gt;, cell_perim_C &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>sholl_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/sholl_data.csv"</span>, <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 599 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): Treatment
dbl  (4): Cohort, Distance(um), avg_intersection_n, intern_C
time (1): Animal_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sholl_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  Animal_ID Cohort Treatment `Distance(um)` avg_intersection_n intern_C
  &lt;time&gt;     &lt;dbl&gt; &lt;chr&gt;              &lt;dbl&gt;              &lt;dbl&gt;    &lt;dbl&gt;
1 01:01          1 Control                5               5.74        6
2 01:01          1 Control               10               7.74        8
3 01:01          1 Control               15              10.2        10
4 01:01          1 Control               20              11.2        11
5 01:01          1 Control               25              12.0        12
6 01:01          1 Control               30              12.4        12</code></pre>
</div>
</div>
</section>
<section id="data-analysisvisualization-replications" class="level2">
<h2 class="anchored" data-anchor-id="data-analysisvisualization-replications">Data Analysis/Visualization Replications</h2>
<ul>
<li>Beta error distributions were used for proportions or percentages bounded between 0 and 1
<ul>
<li>Mean percentage coverage</li>
<li>Percentage coverage</li>
<li>Lacunarity</li>
<li>Density</li>
<li>Circularity</li>
</ul></li>
<li>Negative-binomial error distributions were used for overdispersed integer counts
<ul>
<li>Number of branches</li>
<li>Branch lengths</li>
<li>Number of endpoints</li>
<li>Cell area</li>
<li>Cell perimeter</li>
<li>Mean number of branches</li>
<li>Mean branch length</li>
<li>Mean number of endpoints</li>
<li>Mean number of cells</li>
<li>Sholl mean number of intersections</li>
</ul></li>
<li>Gaussian error distribution was used for a continuous, ~normally distributed variable
<ul>
<li>Fractal dimension</li>
</ul></li>
<li>Gamma error distribution with a log link function was used for a continuous, left-skewed variable
<ul>
<li>Span ratio</li>
</ul></li>
</ul>
<p>Based on the results of previous studies, we expected the effect of distance from cell body on mean number of intersections to be nonlinear. Although to our knowledge no prior studies have explicitly modeled this nonlinearity, we chose to do so to improve inference reliability. We did this by including a natural cubic spline on distance from cell body in the model. In all models for single cell outcome measures, we included random intercepts for individual animal crossed with random intercepts for brain region. This random effects structure appropriately accommodated the hierarchical clustering of multiple data points from each mouse and the hierarchical clustering of data points from each individual within the three brain regions (n = 9 data points within each of three regions from each individual [n = 27 total data points from each mouse]).</p>
<p>In the models for cell-aggregated widefield outcome measures, we included random intercepts for individual mouse crossed with random intercepts for the binned number of cells that were used to obtain the outcome value (n = 9 data points from each mouse). This random effects structure accommodated hierarchical clustering of multiple aggregate data points from each mouse and the possibility that the number of cells used to calculate a given value might introduce unaccounted for variation.</p>
<p>In the Sholl analysis model, we included only a random intercept for individual mouse to accommodate the hierarchical clustering of multiple data points from each mouse (n = 25 distances at which the mean number of intersections was calculated from each mouse).</p>
<p>We fit all models in the frequentist framework using the package glmmTMB in the R statistical computing environment. We based inferences on a combination of coefficient estimates (β) and their 95% confidence intervals, differences between predicted conditional means (∆), effect sizes (d), and p-values following Tukey’s adjustments for multiple comparisons, all of which were obtained using the package emmeans in R.”</p>
</section>
<section id="analysis-1-whole-average-percent-coverage-figure-3a" class="level2">
<h2 class="anchored" data-anchor-id="analysis-1-whole-average-percent-coverage-figure-3a">Analysis 1: Whole Average Percent Coverage (Figure 3a)</h2>
<section id="file-description" class="level3">
<h3 class="anchored" data-anchor-id="file-description">File description</h3>
<p><strong>Perc_Coverage_Avg.csv</strong>: File containing all microglia morphology data collected using percent coverage of Iba1 stain on photomicrographs taken from brain hemispheres.</p>
<ol type="1">
<li><strong>Animal_ID</strong> – Unique identifier for individual mice.</li>
<li><strong>Treatment</strong> – Categorical variable denoting whether each mouse received control diet or Plx5622 (PLX) diet.</li>
<li><strong>Avg_percent</strong> – The percentage of the image covered by dark pixels calculated from nine photomicrographs (3 brain slices per mouse and 3 photomicrographs (of retrosplenial, somatosensory, and entorhinal cortices) per brain slice) of Iba1-stained tissue converted to binary and then averaged to produce a single value per animal.</li>
<li><strong>Perc_Cov</strong> – The averaged percent coverage converted to a decimal.</li>
</ol>
</section>
<section id="data-analysis" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis">Data analysis</h3>
<section id="step-1-exploratory-data-analysis-of-perc_cov-treatment" class="level4">
<h4 class="anchored" data-anchor-id="step-1-exploratory-data-analysis-of-perc_cov-treatment">Step 1: Exploratory data analysis of <strong>Perc_Cov</strong> ~ <strong>Treatment</strong></h4>
<p>Explicit descriptive statistics were not conducted in this paper, but I used <code>skim()</code> to check variable types and calculate five number summaries. I also plotted the spread of the data as a histogram and visualized mean Iba1 coverage (proportion) given Treatment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine number of observations (nrow), number (n_unique) of Treatments</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check variable types and 5 number summaries</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">skim</span>(perc_coverage_avg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Data summary</caption>
<tbody>
<tr class="odd">
<td style="text-align: left;">Name</td>
<td style="text-align: left;">perc_coverage_avg</td>
</tr>
<tr class="even">
<td style="text-align: left;">Number of rows</td>
<td style="text-align: left;">25</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Number of columns</td>
<td style="text-align: left;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">_______________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Column type frequency:</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">character</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">difftime</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">numeric</td>
<td style="text-align: left;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">________________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Group variables</td>
<td style="text-align: left;">None</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: character</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 19%">
<col style="width: 13%">
<col style="width: 19%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 8%">
<col style="width: 12%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">max</th>
<th style="text-align: right;">empty</th>
<th style="text-align: right;">n_unique</th>
<th style="text-align: right;">whitespace</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Treatment</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: difftime</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 18%">
<col style="width: 13%">
<col style="width: 18%">
<col style="width: 13%">
<col style="width: 13%">
<col style="width: 11%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: left;">min</th>
<th style="text-align: left;">max</th>
<th style="text-align: left;">median</th>
<th style="text-align: right;">n_unique</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Animal_ID</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">3660 secs</td>
<td style="text-align: left;">8160 secs</td>
<td style="text-align: left;">02:03:00</td>
<td style="text-align: right;">24</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: numeric</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 16%">
<col style="width: 11%">
<col style="width: 16%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">p0</th>
<th style="text-align: right;">p25</th>
<th style="text-align: right;">p50</th>
<th style="text-align: right;">p75</th>
<th style="text-align: right;">p100</th>
<th style="text-align: left;">hist</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Avg_percent</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">17.74</td>
<td style="text-align: right;">3.26</td>
<td style="text-align: right;">11.64</td>
<td style="text-align: right;">15.01</td>
<td style="text-align: right;">17.83</td>
<td style="text-align: right;">19.8</td>
<td style="text-align: right;">23.24</td>
<td style="text-align: left;">▃▇▇▇▇</td>
</tr>
<tr class="even">
<td style="text-align: left;">Perc_Cov</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.2</td>
<td style="text-align: right;">0.23</td>
<td style="text-align: left;">▃▇▇▇▇</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change Animal_ID variable type from time to character</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>perc_coverage_avg<span class="sc">$</span>Animal_ID <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">strptime</span>(perc_coverage_avg<span class="sc">$</span>Animal_ID, </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">format =</span> <span class="st">"%H:%M:%S"</span>), </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">format =</span> <span class="st">"%H:%M:%S"</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Check Animal_ID is now a character type variable</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(perc_coverage_avg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>spc_tbl_ [25 × 4] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ Animal_ID  : chr [1:25] "01:01:00" "01:05:00" "01:06:00" "01:07:00" ...
 $ Treatment  : chr [1:25] "Control" "Control" "Control" "Control" ...
 $ Avg_percent: num [1:25] 17.5 16.7 12.3 17.4 19.7 ...
 $ Perc_Cov   : num [1:25] 0.175 0.167 0.123 0.174 0.197 ...
 - attr(*, "spec")=
  .. cols(
  ..   Animal_ID = col_time(format = ""),
  ..   Treatment = col_character(),
  ..   Avg_percent = col_double(),
  ..   Perc_Cov = col_double()
  .. )
 - attr(*, "problems")=&lt;externalptr&gt; </code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine how observations are distributed among subjects and treatments</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(perc_coverage_avg, <span class="fu">table</span>(Animal_ID, Treatment))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          Treatment
Animal_ID  Control PLX
  01:01:00       1   0
  01:05:00       1   0
  01:06:00       1   0
  01:07:00       1   0
  01:08:00       1   0
  01:12:00       0   1
  01:13:00       0   1
  01:15:00       0   1
  01:16:00       0   1
  02:01:00       1   1
  02:02:00       1   0
  02:03:00       1   0
  02:04:00       1   0
  02:05:00       1   0
  02:06:00       1   0
  02:07:00       1   0
  02:08:00       1   0
  02:09:00       0   1
  02:11:00       0   1
  02:12:00       0   1
  02:13:00       0   1
  02:14:00       0   1
  02:15:00       0   1
  02:16:00       0   1</code></pre>
</div>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize Perc_Cov given Treatment</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">hist</span>(perc_coverage_avg<span class="sc">$</span>Perc_Cov)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(perc_coverage_avg, <span class="fu">aes</span>(<span class="at">x =</span> Treatment, <span class="at">y =</span> Perc_Cov)) <span class="sc">+</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Treatment),</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.06</span>)) <span class="sc">+</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Treatment), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.3</span>)) <span class="sc">+</span> </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_prism</span>(<span class="st">"floral"</span>) <span class="sc">+</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_prism</span>(<span class="st">"floral"</span>) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">y =</span> <span class="st">"prism_offset_minor"</span>) <span class="sc">+</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel" data-layout="[50,50]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/perc_coverage_avg-1.png" class="img-fluid" width="672"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/perc_coverage_avg-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</section>
<section id="step-2-run-generalized-linear-mixed-model-of-perc_cov-mean-iba1-coverage-given-treatment" class="level4">
<h4 class="anchored" data-anchor-id="step-2-run-generalized-linear-mixed-model-of-perc_cov-mean-iba1-coverage-given-treatment">Step 2: Run generalized linear mixed model of <strong>Perc_Cov</strong> (Mean Iba1 coverage) given <strong>Treatment</strong></h4>
<p>Authors fit all models in the frequentist framework using the package {glmmTMB}.</p>
<blockquote class="blockquote">
<p>NOTE: to fit a model in glmmTMB:</p>
</blockquote>
<blockquote class="blockquote">
<ul>
<li>specify a model for the conditional effects, in the standard R formula notation</li>
<li>specify a model for the random effects, in the notation that is common to the {nlme} and {lme4} packages. Random effects are specified as x|g, where x is an effect and g is a grouping factor (which must be a factor variable, or a nesting of/interaction among factor variables). Formula examples:</li>
<li>1|block for a random-intercept model</li>
<li>A model of crossed random effects (block and year) would be (1|block)+(1|year).</li>
<li>choose the error distribution by specifying the family (family argument). Specify the function</li>
<li>binomial, gaussian, poisson, Gamma from base R</li>
<li>nbinom2, beta family(), betabinomial from family glmmTMB</li>
<li>choose the error distribution by specifying the family (family argument).</li>
<li>Distributions defined in base R and documented in ?family</li>
<li>Distributions defined in ?glmmTMB::family glmmTMB</li>
<li>optionally specify a zero-inflation model (ziformula argument) with fixed and/or random effects</li>
<li>optionally specify a dispersion model with fixed effects</li>
</ul>
</blockquote>
<p>Components of generalized linear mixed model of <strong>Perc_Cov</strong> given <strong>Treatment</strong></p>
<ul>
<li>Fixed effect for <strong>Treatment</strong></li>
<li>Beta error distribution, as <strong>Perc_Cov</strong> values are bounded between 0 and 1</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># model</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>perc_cov_avg_model <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(Perc_Cov <span class="sc">~</span> Treatment, </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">family =</span> <span class="fu">beta_family</span>(<span class="at">link =</span> <span class="st">"logit"</span>),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data =</span> perc_coverage_avg)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(perc_cov_avg_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Family: beta  ( logit )
Formula:          Perc_Cov ~ Treatment
Data: perc_coverage_avg

     AIC      BIC   logLik deviance df.resid 
   -95.4    -91.8     50.7   -101.4       22 


Dispersion parameter for beta family ():  141 

Conditional model:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  -1.57444    0.06159  -25.56   &lt;2e-16 ***
TreatmentPLX  0.08311    0.08746    0.95    0.342    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Formula <span class="math display">\[logit(\pi_i) = -1.57444 + 0.08311 \times {Treatment}\]</span></p>
</section>
<section id="step-3-obtain-inferential-statistics-using-emmeans" class="level4">
<h4 class="anchored" data-anchor-id="step-3-obtain-inferential-statistics-using-emmeans">Step 3: Obtain inferential statistics using {emmeans}</h4>
<p>Authors used {emmeans} to obtain coefficient estimates (β) and their 95% confidence intervals, differences between predicted conditional means (∆), effect sizes (d), and p-values following Tukey’s adjustments for multiple comparisons</p>
<p>Differences between predicted conditional means (∆) are reported in %, which I interpreted to be the absolute value in units of the response variable (mean Iba1 coverage). I reported my estimate of the difference in units of proportion.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Model converted to log scale, want results on response scale</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get point estimates (estimated marginal means) and 95% CIs </span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>emm_r <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(perc_cov_avg_model, <span class="sc">~</span> Treatment,  <span class="at">component =</span> <span class="st">"response"</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the point estimates and their 95% confidence intervals</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>(estimates_r <span class="ot">&lt;-</span> <span class="fu">summary</span>(emm_r, <span class="at">type =</span> <span class="st">"response"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Treatment emmean      SE  df asymp.LCL asymp.UCL
 Control    0.172 0.00875 Inf     0.154     0.189
 PLX        0.184 0.00936 Inf     0.165     0.202

Confidence level used: 0.95 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the pairwise comparisons</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate = difference between predicted conditional means (∆) and p-value</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># why is estimate negative??</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>(pairwise_r <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_r, <span class="at">adjust =</span> <span class="st">"tukey"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      estimate     SE  df z.ratio p.value
 Control - PLX  -0.0121 0.0128 Inf  -0.949  0.3424</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To obtain effect size using pooled standard deviation from model, </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co"># need difference between means on log scale</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(perc_cov_avg_model, <span class="sc">~</span> Treatment)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>pairwise <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm, <span class="at">adjust =</span> <span class="st">"tukey"</span>)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>(differences <span class="ot">&lt;-</span> <span class="fu">summary</span>(pairwise))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      estimate     SE  df z.ratio p.value
 Control - PLX  -0.0831 0.0875 Inf  -0.950  0.3420

Results are given on the log odds ratio (not the response) scale. </code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># effect size (d)</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># d = Difference between means / Pooled standard deviation</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>(effect_size <span class="ot">&lt;-</span> <span class="fu">eff_size</span>(emm, </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sigma =</span> <span class="fu">sigma</span>(perc_cov_avg_model),</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">edf =</span> <span class="fu">df.residual</span>(perc_cov_avg_model)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      effect.size       SE  df asymp.LCL asymp.UCL
 Control - PLX    -0.00059 0.000627 Inf  -0.00182  0.000639

sigma used for effect sizes: 140.9 
Confidence level used: 0.95 </code></pre>
</div>
</div>
<section id="step-4-visualize-mean-iba1-coverage-treatment" class="level5">
<h5 class="anchored" data-anchor-id="step-4-visualize-mean-iba1-coverage-treatment">Step 4: Visualize Mean Iba1 coverage ~ treatment</h5>
<p>Scatter dot plot with estimated means and 95% CI</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set plot size</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width =</span> <span class="dv">2</span>, <span class="at">repr.plot.height =</span><span class="dv">4</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co"># define custom graph colors</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>gray_scale <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#A0A0A4"</span>, <span class="st">"#808080"</span>)</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(perc_coverage_avg, <span class="fu">aes</span>(<span class="at">x =</span> Treatment)) <span class="sc">+</span> </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> Perc_Cov, <span class="at">color =</span> <span class="fu">factor</span>(Treatment)),</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.06</span>),</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span> </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="at">data =</span> estimates_r,</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">y =</span> emmean, <span class="at">ymin =</span> asymp.LCL, <span class="at">ymax =</span> asymp.UCL),</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="fl">0.4</span>) <span class="sc">+</span>  <span class="co"># Add point estimates and confidence intervals</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> estimates_r, <span class="fu">aes</span>(<span class="at">y =</span> emmean, <span class="at">ymin =</span> asymp.LCL, <span class="at">ymax =</span> asymp.UCL), </span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), </span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.4</span>,</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">linewidth =</span> <span class="fl">0.7</span>) <span class="sc">+</span> <span class="co"># Add confidence intervals bars</span></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.3</span>)) <span class="sc">+</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Mean Iba1 Coverage (%)"</span>) <span class="sc">+</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Treatment"</span>)) <span class="sc">+</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>)</span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Need to update p.value by hand</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a><span class="co"># add p_values</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>p_vals <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>group1, <span class="sc">~</span>group2, <span class="sc">~</span>p.value,   <span class="sc">~</span>y.position,</span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Control"</span>,   <span class="st">"PLX"</span>,     <span class="fl">0.34</span>, <span class="fl">0.295</span></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> base <span class="sc">+</span> </span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> gray_scale) <span class="sc">+</span></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>(<span class="at">base_size =</span> <span class="dv">16</span>,</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">base_fontface =</span> <span class="st">"plain"</span>) <span class="sc">+</span> </span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb35-36"><a href="#cb35-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.2</span>),</span>
<span id="cb35-37"><a href="#cb35-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.2</span>),</span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust=</span><span class="fl">1.5</span>), <span class="co"># move labels closer to axis</span></span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_pvalue</span>(p_vals, <span class="at">label =</span> <span class="st">"p = {p.value}"</span>, <span class="at">tip.length =</span> <span class="dv">0</span>, <span class="at">label.size =</span> <span class="dv">4</span>, <span class="at">vjust=</span><span class="sc">-</span><span class="fl">0.3</span>) <span class="co"># added p-value</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Save plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"ERC_fig3a.png"</span>, </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="fl">3.5</span>, <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">path =</span> <span class="st">"images/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="comparison-of-results" class="level3">
<h3 class="anchored" data-anchor-id="comparison-of-results">Comparison of Results</h3>
<div>

</div>
<div class="quarto-layout-panel" data-layout="[50,50]">
<div class="quarto-layout-row quarto-layout-valign-bottom">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/ERC_fig3a.png" class="img-fluid figure-img"></p>
<figcaption>Mine: p-value = 0.34, Mean ∆ = -0.012, d = -0.0006</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/fig3a.png" class="img-fluid figure-img"></p>
<figcaption>Paper: p-value = 0.35, Mean ∆ = 1.2%, d = 0.0006</figcaption>
</figure>
</div>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Figure 3a. Full photomicrograph analyses of microglial morphology were not sensitive to differences between control and treatments groups. Mean percent coverage of Iba1 staining per animal (3 brain slices per mouse and 3 photomicrographs per brain slice were used) is shown as individual data points. Treatment n = 12 mice, Control n = 13 mice. Results are presented as point estimates with 95% confidence intervals, which were estimated from mixed effects model.</p>
</blockquote>
<blockquote class="blockquote">
<p>NOTE: Mean Iba1 coverage analyzed/graphed as <em>proportion</em> of coverage, while mean ∆ appears to be reported by authors as |<em>percentage</em> of coverage|. Figure caption specifies negative-binomial error distribution, but statistical analysis section specifies average percent coverage was modeled with beta error distribution.</p>
</blockquote>
<div>

</div>
<div class="cell quarto-layout-panel" data-layout="[55,45]">

</div>
</section>
</section>
<section id="analysis-2-whole-percent-coverage" class="level2">
<h2 class="anchored" data-anchor-id="analysis-2-whole-percent-coverage">Analysis 2: Whole Percent Coverage</h2>
<section id="file-description-1" class="level3">
<h3 class="anchored" data-anchor-id="file-description-1">File description</h3>
<p><strong>Perc_Coverage.csv</strong>: File containing all microglia morphology data collected using percent coverage of Iba1 stain on photomicrographs taken from brain hemispheres.</p>
<ol type="1">
<li><strong>Animal_ID</strong> – Unique identifier for individual mice.</li>
<li><strong>Treatment</strong> – Categorical variable denoting whether each mouse received c ontrol diet or Plx5622 (PLX) diet.</li>
<li><strong>Coverage</strong> – The percentage of the image covered by dark pixels calculated from photomicrographs of Iba1-stained tissue converted to binary.</li>
<li><strong>Perc_Cov</strong> – The percent coverage converted to a decimal.</li>
</ol>
</section>
<section id="data-wrangling" class="level3">
<h3 class="anchored" data-anchor-id="data-wrangling">Data wrangling</h3>
<p>Clean up data frame for hierarchical generalized linear mixed model</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change Animal_ID variable type from time to character</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>perc_coverage<span class="sc">$</span>Animal_ID <span class="ot">&lt;-</span> <span class="fu">format</span>(<span class="fu">strptime</span>(perc_coverage<span class="sc">$</span>Animal_ID, </span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">format =</span> <span class="st">"%H:%M:%S"</span>), </span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">format =</span> <span class="st">"%H:%M:%S"</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check binned number of cells per subject in perc_coverage</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> perc_coverage <span class="sc">%&gt;%</span> </span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Animal_ID, Treatment) <span class="sc">%&gt;%</span> </span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">binned_cells =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'Animal_ID'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add binned number of cells to perc_coverage</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>perc_coverage <span class="ot">&lt;-</span> <span class="fu">full_join</span>(perc_coverage, cells)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(Animal_ID, Treatment)`</code></pre>
</div>
</div>
</section>
<section id="data-analysis-1" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-1">Data Analysis</h3>
<section id="step-1-exploratory-data-analysis-of-perc_cov-treatment-1" class="level4">
<h4 class="anchored" data-anchor-id="step-1-exploratory-data-analysis-of-perc_cov-treatment-1">Step 1: Exploratory data analysis of <strong>Perc_Cov</strong> ~ <strong>Treatment</strong></h4>
<p>Explicit descriptive statistics were not conducted in this paper, but I used <code>skim()</code> to check variable types and calculate five number summaries. I also plotted the spread of the data as a histogram and visualized Iba1 coverage (proportion) given Treatment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine number of observations (nrow), number (n_unique) of Treatments</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Check variable types and 5 number summaries</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">skim</span>(perc_coverage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Data summary</caption>
<tbody>
<tr class="odd">
<td style="text-align: left;">Name</td>
<td style="text-align: left;">perc_coverage</td>
</tr>
<tr class="even">
<td style="text-align: left;">Number of rows</td>
<td style="text-align: left;">225</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Number of columns</td>
<td style="text-align: left;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">_______________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Column type frequency:</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">character</td>
<td style="text-align: left;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">numeric</td>
<td style="text-align: left;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">________________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Group variables</td>
<td style="text-align: left;">None</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: character</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 19%">
<col style="width: 13%">
<col style="width: 19%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 8%">
<col style="width: 12%">
<col style="width: 15%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">max</th>
<th style="text-align: right;">empty</th>
<th style="text-align: right;">n_unique</th>
<th style="text-align: right;">whitespace</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Animal_ID</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">Treatment</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: numeric</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 16%">
<col style="width: 11%">
<col style="width: 16%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">p0</th>
<th style="text-align: right;">p25</th>
<th style="text-align: right;">p50</th>
<th style="text-align: right;">p75</th>
<th style="text-align: right;">p100</th>
<th style="text-align: left;">hist</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Coverage</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">17.74</td>
<td style="text-align: right;">4.59</td>
<td style="text-align: right;">6.59</td>
<td style="text-align: right;">14.89</td>
<td style="text-align: right;">17.92</td>
<td style="text-align: right;">21.12</td>
<td style="text-align: right;">29.14</td>
<td style="text-align: left;">▂▅▇▆▁</td>
</tr>
<tr class="even">
<td style="text-align: left;">Perc_Cov</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">0.15</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.21</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: left;">▂▅▇▆▁</td>
</tr>
<tr class="odd">
<td style="text-align: left;">binned_cells</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: left;">▁▁▇▁▁</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine how observations are distributed among subjects and treatments</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(perc_coverage, <span class="fu">table</span>(Animal_ID, Treatment))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          Treatment
Animal_ID  Control PLX
  01:01:00       9   0
  01:05:00       9   0
  01:06:00       9   0
  01:07:00       9   0
  01:08:00       9   0
  01:12:00       0   9
  01:13:00       0   9
  01:15:00       0   9
  01:16:00       0   9
  02:01:00       9   9
  02:02:00       9   0
  02:03:00       9   0
  02:04:00       9   0
  02:05:00       9   0
  02:06:00       9   0
  02:07:00       9   0
  02:08:00       9   0
  02:09:00       0   9
  02:11:00       0   9
  02:12:00       0   9
  02:13:00       0   9
  02:14:00       0   9
  02:15:00       0   9
  02:16:00       0   9</code></pre>
</div>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize Perc_Cov given Treatment</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">hist</span>(perc_coverage<span class="sc">$</span>Perc_Cov)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(perc_coverage, <span class="fu">aes</span>(<span class="at">x =</span> Treatment, <span class="at">y =</span> Perc_Cov)) <span class="sc">+</span> </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Treatment),</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.07</span>)) <span class="sc">+</span> </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Treatment), <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.3</span>)) <span class="sc">+</span> </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_prism</span>(<span class="st">"floral"</span>) <span class="sc">+</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_prism</span>(<span class="st">"floral"</span>) <span class="sc">+</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">y =</span> <span class="st">"prism_offset_minor"</span>) <span class="sc">+</span> </span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span> </span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel" data-layout="[50,50]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/perc_coverage-1.png" class="img-fluid" width="672"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="images/perc_coverage-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</div>
</section>
<section id="step-2-run-hierarchical-generalized-linear-mixed-model-of-perc_cov-iba1-coverage-given-treatment" class="level4">
<h4 class="anchored" data-anchor-id="step-2-run-hierarchical-generalized-linear-mixed-model-of-perc_cov-iba1-coverage-given-treatment">Step 2: Run hierarchical generalized linear mixed model of <strong>Perc_Cov</strong> (Iba1 coverage) given <strong>Treatment</strong></h4>
<ul>
<li>Include fixed effect for <strong>Treatment</strong></li>
<li>Include random intercepts for <strong>Animal_ID</strong>, individual mouse, <em>crossed with</em> random intercepts for <strong>binned_cells</strong>, the binned number of cells that were used to obtain the outcome value (n = 9 data points per mouse).
<ul>
<li>Random effects structure accommodated…</li>
<li>hierarchical clustering of multiple aggregate data points from each mouse and</li>
<li>the possibility that the number of cells used to calculate a given value might introduce unaccounted for variation.</li>
</ul></li>
<li>Beta error distribution, as <strong>Perc_Cov</strong> values are bounded between 0 and 1</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>perc_cov_model <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(Perc_Cov <span class="sc">~</span> Treatment <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Animal_ID) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>binned_cells),</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">family =</span> <span class="fu">beta_family</span>(<span class="at">link=</span><span class="st">"logit"</span>),</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data =</span> perc_coverage)</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(perc_cov_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Family: beta  ( logit )
Formula:          Perc_Cov ~ Treatment + (1 | Animal_ID) + (1 | binned_cells)
Data: perc_coverage

     AIC      BIC   logLik deviance df.resid 
  -796.2   -779.1    403.1   -806.2      220 

Random effects:

Conditional model:
 Groups       Name        Variance  Std.Dev. 
 Animal_ID    (Intercept) 4.738e-02 2.177e-01
 binned_cells (Intercept) 1.473e-10 1.214e-05
Number of obs: 225, groups:  Animal_ID, 24; binned_cells, 1

Dispersion parameter for beta family ():  107 

Conditional model:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  -1.65878    0.06413  -25.86  &lt; 2e-16 ***
TreatmentPLX  0.24164    0.08786    2.75  0.00595 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Random intercepts do not seem to add explanatory power (very low amount of Variance in response variable explained)</p>
<p>Formula <span class="math display">\[logit(\pi_i) = -1.65878 + 0.24164 \times {Treatment}\]</span></p>
</section>
<section id="step-3-obtain-inferential-statistics-using-emmeans-1" class="level4">
<h4 class="anchored" data-anchor-id="step-3-obtain-inferential-statistics-using-emmeans-1">Step 3: Obtain inferential statistics using {emmeans}</h4>
<p>Authors used {emmeans} to obtain coefficient estimates (β) and their 95% confidence intervals, differences between predicted conditional means (∆), effect sizes (d), and p-values following Tukey’s adjustments for multiple comparisons</p>
<p>Differences between predicted conditional means (∆) are reported in %, which I interpreted to be the absolute value in units of the response variable (mean Iba1 coverage). I reported my estimate of the difference in units of proportion.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Model converted to log scale, want results on response scale</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get point estimates (estimated marginal means) and 95% CIs </span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>emm_r <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(perc_cov_model, <span class="sc">~</span> Treatment,  <span class="at">component =</span> <span class="st">"response"</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the point estimates and their 95% confidence intervals</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>(estimates_r <span class="ot">&lt;-</span> <span class="fu">summary</span>(emm_r, <span class="at">type =</span> <span class="st">"response"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Treatment emmean      SE  df asymp.LCL asymp.UCL
 Control    0.160 0.00862 Inf     0.143     0.177
 PLX        0.195 0.01027 Inf     0.175     0.215

Confidence level used: 0.95 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the pairwise comparisons</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co"># estimate = difference between predicted conditional means (∆) and p-value</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="co"># why is estimate negative??</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>(pairwise_r <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm_r, <span class="at">adjust =</span> <span class="st">"tukey"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      estimate     SE  df z.ratio p.value
 Control - PLX  -0.0352 0.0129 Inf  -2.734  0.0062</code></pre>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To obtain effect size using pooled standard deviation from model, </span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="co"># need difference between means on log scale</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>emm <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(perc_cov_model, <span class="sc">~</span> Treatment)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>pairwise <span class="ot">&lt;-</span> <span class="fu">pairs</span>(emm, <span class="at">adjust =</span> <span class="st">"tukey"</span>)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>(differences <span class="ot">&lt;-</span> <span class="fu">summary</span>(pairwise))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      estimate     SE  df z.ratio p.value
 Control - PLX   -0.242 0.0879 Inf  -2.750  0.0060

Results are given on the log odds ratio (not the response) scale. </code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># effect size (d)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="co"># d = Difference between means / Pooled standard deviation</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>(effect_size <span class="ot">&lt;-</span> <span class="fu">eff_size</span>(emm, </span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sigma =</span> <span class="fu">sigma</span>(perc_cov_model),</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">edf =</span> <span class="fu">df.residual</span>(perc_cov_model)</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>                         )</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> contrast      effect.size       SE  df asymp.LCL asymp.UCL
 Control - PLX    -0.00226 0.000827 Inf  -0.00388 -0.000635

sigma used for effect sizes: 107.1 
Confidence level used: 0.95 </code></pre>
</div>
</div>
<section id="step-4-visualize-iba1-coverage-treatment" class="level5">
<h5 class="anchored" data-anchor-id="step-4-visualize-iba1-coverage-treatment">Step 4: Visualize Iba1 coverage ~ treatment</h5>
<p>Scatter dot plot with estimated means and 95% CI</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set plot size</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">repr.plot.width =</span> <span class="dv">2</span>, <span class="at">repr.plot.height =</span><span class="dv">4</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="co"># define custom graph colors</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>gray_scale <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#A0A0A4"</span>, <span class="st">"#808080"</span>)</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(perc_coverage, <span class="fu">aes</span>(<span class="at">x =</span> Treatment)) <span class="sc">+</span> </span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> Perc_Cov, <span class="at">color =</span> <span class="fu">factor</span>(Treatment)),</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.07</span>),</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span> </span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="at">data =</span> estimates_r,</span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">y =</span> emmean, <span class="at">ymin =</span> asymp.LCL, <span class="at">ymax =</span> asymp.UCL),</span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">size =</span> <span class="fl">0.4</span>) <span class="sc">+</span>  <span class="co"># Add point estimates and confidence intervals</span></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> estimates_r, <span class="fu">aes</span>(<span class="at">y =</span> emmean, <span class="at">ymin =</span> asymp.LCL, <span class="at">ymax =</span> asymp.UCL), </span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), </span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.4</span>,</span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">linewidth =</span> <span class="fl">0.7</span>) <span class="sc">+</span> <span class="co"># Add confidence intervals bars</span></span>
<span id="cb55-19"><a href="#cb55-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.0</span>, <span class="fl">0.3</span>)) <span class="sc">+</span></span>
<span id="cb55-20"><a href="#cb55-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Iba1 Coverage (%)"</span>) <span class="sc">+</span></span>
<span id="cb55-21"><a href="#cb55-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels=</span><span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Treatment"</span>)) <span class="sc">+</span></span>
<span id="cb55-22"><a href="#cb55-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>)</span>
<span id="cb55-23"><a href="#cb55-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-24"><a href="#cb55-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Need to update p.value by hand</span></span>
<span id="cb55-25"><a href="#cb55-25" aria-hidden="true" tabindex="-1"></a><span class="co"># add p_values</span></span>
<span id="cb55-26"><a href="#cb55-26" aria-hidden="true" tabindex="-1"></a>p_vals <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb55-27"><a href="#cb55-27" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>group1, <span class="sc">~</span>group2, <span class="sc">~</span>p.value,   <span class="sc">~</span>y.position,</span>
<span id="cb55-28"><a href="#cb55-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Control"</span>,   <span class="st">"PLX"</span>,     <span class="fl">0.006</span>, <span class="fl">0.295</span></span>
<span id="cb55-29"><a href="#cb55-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-30"><a href="#cb55-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-31"><a href="#cb55-31" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> base <span class="sc">+</span> </span>
<span id="cb55-32"><a href="#cb55-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> gray_scale) <span class="sc">+</span></span>
<span id="cb55-33"><a href="#cb55-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>(<span class="at">base_size =</span> <span class="dv">16</span>,</span>
<span id="cb55-34"><a href="#cb55-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">base_fontface =</span> <span class="st">"plain"</span>) <span class="sc">+</span> </span>
<span id="cb55-35"><a href="#cb55-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb55-36"><a href="#cb55-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.2</span>),</span>
<span id="cb55-37"><a href="#cb55-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">linewidth =</span> <span class="fl">0.2</span>),</span>
<span id="cb55-38"><a href="#cb55-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb55-39"><a href="#cb55-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust=</span><span class="fl">1.5</span>), <span class="co"># move labels closer to axis</span></span>
<span id="cb55-40"><a href="#cb55-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb55-41"><a href="#cb55-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_pvalue</span>(p_vals, <span class="at">label =</span> <span class="st">"p = {p.value}"</span>, <span class="at">tip.length =</span> <span class="dv">0</span>, <span class="at">label.size =</span> <span class="dv">4</span>, <span class="at">vjust=</span><span class="sc">-</span><span class="fl">0.3</span>) <span class="co"># added p-value</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Save plot</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"ERC_fig3b.png"</span>, </span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="fl">3.5</span>, <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">path =</span> <span class="st">"images/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="comparison-of-results-1" class="level3">
<h3 class="anchored" data-anchor-id="comparison-of-results-1">Comparison of Results</h3>
<div>

</div>
<div class="quarto-layout-panel" data-layout="[50,50]">
<div class="quarto-layout-row quarto-layout-valign-bottom">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/ERC_fig3b.png" class="img-fluid figure-img"></p>
<figcaption>Mine: p-value = 0.006, Mean ∆ = -0.035, d = -0.002</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/fig3b.png" class="img-fluid figure-img"></p>
<figcaption>Paper: p-value = 0.006, Mean ∆ = 3.5%, d = 0.002</figcaption>
</figure>
</div>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Figure 3b. Photomicrograph-specific (unaveraged) percent coverage detected significantly more Iba1 staining in the treatment group compared to controls. Percent coverage of Iba1 staining, with multiple data points per subject, is shown as individual data points. Treatment n = 108 images (12 mice, 9 images/mouse), Control n = 117 images (13 mice, 9 images/mouse). Results are presented as point estimates with 95% confidence intervals, which were estimated from mixed effects model.</p>
</blockquote>
<blockquote class="blockquote">
<p>NOTE: Mean Iba1 coverage analyzed/graphed as <em>proportion</em> of coverage, while mean ∆ appears to be reported by authors as |<em>percentage</em> of coverage|. Figure caption specifies negative-binomial error distribution, but statistical analysis section specifies percent coverage was modeled with beta error distribution.</p>
</blockquote>
</section>
</section>
<section id="analysis-3-whole-skeletal-analysis" class="level2">
<h2 class="anchored" data-anchor-id="analysis-3-whole-skeletal-analysis">Analysis 3: Whole Skeletal Analysis</h2>
<section id="file-description-2" class="level3">
<h3 class="anchored" data-anchor-id="file-description-2">File description</h3>
<p>Not available?</p>
</section>
<section id="data-analysis-2" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-2">Data analysis</h3>
</section>
<section id="result" class="level3">
<h3 class="anchored" data-anchor-id="result">Result</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="images/fig3c.png" class="img-fluid figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p>“Full photomicrograph skeletal analysis only detected that microglia in the treatment group had fewer endpoints than controls. Mean microglial branch length was calculated by averaging the total length of branches in the photomicrograph by the total number of microglia in the photomicrograph. Full photomicrograph skeletal analysis did not detect differences in mean microglial branch length between photomicrographs from the treatment groups (p-value = 0.40, Mean ∆ = 28 μm, d = 0.002 Fig. 3c). Photomicrographs from the treatment group indicated that microglia had fewer mean endpoints than control microglia (p-value = 0.002, Mean ∆ = 18, d = 0.01; Fig. 3d). A difference in mean number of microglia per photomicrograph between the treatment groups was not detected (p-value = 0.60, Mean ∆ = 0, d = 0.01; Fig. 3e). A difference in mean microglial branches per cell per photomicrograph between treatment groups was not detected (p-value = 0.08, Mean ∆ = 13, d = 0.005; Fig. 3f).”</p>
<p>“Figure 3. Full photomicrograph skeletal analysis detected that microglia from the treatment group had fewer endpoints than the controls but did not detect differences in branch number or branch length. (c) Mean microglial branch length; (d) mean number of endpoints per cell per photomicrograph; (e) mean number of microglial cells per photomicrograph; and (f) mean number of microglial branches per cell per image. Individual data points are presented as gray dots. (d–f) Treatment n = 108 images (12 mice, 9 images/mouse), Control n = 117 images (13 mice, 9 images/mouse). Results are presented as point estimates with 95% confidence intervals, which were estimated from mixed effects models with a negative-binomial error distribution.”</p>
</section>
</section>
<section id="analysis-4-single-cell-fractal-analysis" class="level2">
<h2 class="anchored" data-anchor-id="analysis-4-single-cell-fractal-analysis">Analysis 4: Single Cell Fractal Analysis</h2>
<section id="file-description-3" class="level3">
<h3 class="anchored" data-anchor-id="file-description-3">File description</h3>
<p><strong>Single_Cell.csv</strong>: File containing all microglia morphology data collected using single cell skeletal analysis and fractal analysis of Iba1-stained microglia randomly selected and isolated from photomicrographs taken from brain hemispheres.</p>
<ol type="1">
<li><strong>Animal_ID</strong> – Unique identifier for individual mice.</li>
<li><strong>Cohort</strong> – Categorical variable denoting the experimental cohort each mouse was in.</li>
<li><strong>Treatment</strong> – Categorical variable denoting whether each mouse received control diet or Plx5622 (PLX) diet.</li>
<li><strong>Region</strong> – Categorical variable denoting the brain region from which each photomicrograph was taken.</li>
</ol>
<ul>
<li>Region 1: entorhinal cortex</li>
<li>Region 2: primary somatosensory barrel field cortex</li>
<li>Region 3: retrosplenial cortex.</li>
</ul>
<ol start="5" type="1">
<li><strong>FractalDimension(DB</strong>) – Continuous variable calculated with fractal analysis of a statistical measure of pattern complexity of a microglial cell (Db).</li>
<li><strong>Lacunarity</strong> – Continuous variable calculated with fractal analysis of a geomatric measure of how the microglial cell/pattern fills space.</li>
<li><strong>Density(pixels/area)</strong> – Continuous variable calculated with fractal analysis of the number of pixels/area occupied by the microglial cell.</li>
<li><strong>SpanRatio</strong> – Continuous variable calculated with fractal analysis of the longest length/longest width of the microglial cell.</li>
<li><strong>Circularity</strong> – Continuous variable calculated with fractal analysis of how circular the microglia cell was.</li>
<li><strong>NumberOfEndpoints</strong> – Continuous variable of Iba1 microglia end points (n).</li>
<li><strong>BranchLength(um)</strong> – Continuous variable of Iba1 microglia branch lengths (μm).</li>
<li><strong>NumberBranches</strong> – Continuous variable of Iba1 microglia branches (n).</li>
<li><strong>body_area</strong> – Continuous variable of cell body area of microglia.</li>
<li><strong>cell_area_C</strong> – Continuous variable of cell body area of microglia rounded to the nearest whole number.</li>
<li><strong>body_perimeter</strong> – Continuous variable of cell body perimeter of microglia (μm).</li>
<li><strong>cell_perim_C</strong> – Continuous variable of cell body perimeter of microglia rounded to the nearest whole number (μm^2).</li>
</ol>
</section>
<section id="data-analysis-3" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-3">Data analysis</h3>
</section>
<section id="result-1" class="level3">
<h3 class="anchored" data-anchor-id="result-1">Result</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="images/fig4a.png" class="img-fluid figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p>“Fractal analysis detected that microglia from the treatment group had a less complex branching pattern than controls. Fractal analysis performed on isolated microglia detected that microglia from the treatment group had a less complex branching pattern (fractal dimension) than controls (p-value &lt; 0.0001, Mean ∆ = 0.05, d = 0.65; Fig. 4a). However, differences in lacunarity (p-value = 0.72, Mean ∆ = 0.013, d = 0.004), circularity (p-value = 0.08, Mean ∆ = 0.013, d = 0.003), span ratio (p-value = 0.33, Mean ∆ = 0.03,d = 0.10), and density (p-value = 0.22, Mean ∆ = 0.004, d = 0.0001) between treatment groups were not detected (Fig. 4b–e).”</p>
</section>
</section>
<section id="analysis-5-single-cell-skeletal-analysis" class="level2">
<h2 class="anchored" data-anchor-id="analysis-5-single-cell-skeletal-analysis">Analysis 5: Single Cell Skeletal Analysis</h2>
<section id="file-description-4" class="level3">
<h3 class="anchored" data-anchor-id="file-description-4">File description</h3>
<p><strong>Single_Cell.csv</strong>: File containing all microglia morphology data collected using single cell skeletal analysis and fractal analysis of Iba1-stained microglia randomly selected and isolated from photomicrographs taken from brain hemispheres.</p>
<ol type="1">
<li><strong>Animal_ID</strong> – Unique identifier for individual mice.</li>
<li><strong>Cohort</strong> – Categorical variable denoting the experimental cohort each mouse was in.</li>
<li><strong>Treatment</strong> – Categorical variable denoting whether each mouse received control diet or Plx5622 (PLX) diet.</li>
<li><strong>Region</strong> – Categorical variable denoting the brain region from which each photomicrograph was taken.</li>
</ol>
<ul>
<li>Region 1: entorhinal cortex</li>
<li>Region 2: primary somatosensory barrel field cortex</li>
<li>Region 3: retrosplenial cortex.</li>
</ul>
<ol start="5" type="1">
<li><strong>FractalDimension(DB</strong>) – Continuous variable calculated with fractal analysis of a statistical measure of pattern complexity of a microglial cell (Db).</li>
<li><strong>Lacunarity</strong> – Continuous variable calculated with fractal analysis of a geomatric measure of how the microglial cell/pattern fills space.</li>
<li><strong>Density(pixels/area)</strong> – Continuous variable calculated with fractal analysis of the number of pixels/area occupied by the microglial cell.</li>
<li><strong>SpanRatio</strong> – Continuous variable calculated with fractal analysis of the longest length/longest width of the microglial cell.</li>
<li><strong>Circularity</strong> – Continuous variable calculated with fractal analysis of how circular the microglia cell was.</li>
<li><strong>NumberOfEndpoints</strong> – Continuous variable of Iba1 microglia end points (n).</li>
<li><strong>BranchLength(um)</strong> – Continuous variable of Iba1 microglia branch lengths (μm).</li>
<li><strong>NumberBranches</strong> – Continuous variable of Iba1 microglia branches (n).</li>
<li><strong>body_area</strong> – Continuous variable of cell body area of microglia.</li>
<li><strong>cell_area_C</strong> – Continuous variable of cell body area of microglia rounded to the nearest whole number.</li>
<li><strong>body_perimeter</strong> – Continuous variable of cell body perimeter of microglia (μm).</li>
<li><strong>cell_perim_C</strong> – Continuous variable of cell body perimeter of microglia rounded to the nearest whole number (μm^2).</li>
</ol>
</section>
<section id="data-analysis-4" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-4">Data analysis</h3>
</section>
<section id="result-2" class="level3">
<h3 class="anchored" data-anchor-id="result-2">Result</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="images/fig4f.png" class="img-fluid figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p>“Single cell skeletal analysis detected that microglia from the treatment group had larger cell bodies, fewer and shorter branches, and fewer endpoints than controls. Microglia from the treatment group had a larger cell body perimeter (p-value = 0.002, Mean ∆ = 14 μm, d = 0.01; Fig. 4f) and area (p-value = 0.01, Mean ∆ = 148 μm2, d = 0.02; Fig. 4g) than controls. Microglia from the treatment group had fewer branches per cell (p-value = 0.002, Mean ∆ = 52, d = 0.04; Fig. 4h), shorter branch length (p-value &lt; 0.0001, Mean ∆ = 315 μm, d = 0.03; Fig. 4i), and fewer endpoints per cell (p-value = 0.001, Mean ∆ = 23, d = 0.03; Fig. 4j) than controls.”</p>
</section>
</section>
<section id="analysis-6-single-cell-sholl-analysis" class="level2">
<h2 class="anchored" data-anchor-id="analysis-6-single-cell-sholl-analysis">Analysis 6: Single Cell Sholl Analysis</h2>
<section id="file-description-5" class="level3">
<h3 class="anchored" data-anchor-id="file-description-5">File description</h3>
<p><strong>Sholl_Data.csv</strong>: File containing all microglia morphology data collected using Sholl analysis of Iba1-stained microglia randomly selected and isolated from photomicrographs taken from brain hemispheres.</p>
<ol type="1">
<li><strong>Animal_ID</strong> – Unique identifier for individual mice.</li>
<li><strong>Cohort</strong> – Categorical variable denoting the experimental cohort each mouse was in.</li>
<li><strong>Treatment</strong> – Categorical variable denoting whether each mouse received control diet or Plx5622 (PLX) diet.</li>
<li><strong>Distance(um)</strong> – Continuous variable of the Sholl circle distance from the microglial cell soma (μm).</li>
<li><strong>avg_intersection_n</strong> – Continuous variable of the average number of times a microglial branch intersected a Sholl circle.</li>
<li><strong>intern_C</strong> – Continuous variable of the average number of times a microglial branch intersected a Sholl circle rounded to the nearest whole number.</li>
</ol>
</section>
<section id="data-analysis-5" class="level3">
<h3 class="anchored" data-anchor-id="data-analysis-5">Data analysis</h3>
<p>In the Sholl analysis, we also included a fixed effect for distance from cell body as well as a two-way interaction between treatment and distance from cell body.</p>
</section>
<section id="result-3" class="level3">
<h3 class="anchored" data-anchor-id="result-3">Result</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="images/fig4k.png" class="img-fluid figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p>“Sholl analysis detected that microglia from the treatment group had fewer and less expansive branches than controls. Microglia from the treatment group had fewer branches intersecting the Sholl analysis circles than controls (p-value = 0.002, Mean ∆ = 2.29 intersections, d = 0.003; Fig. 4k).”</p>
</section>
</section>
<section id="discussion-and-reflection" class="level2">
<h2 class="anchored" data-anchor-id="discussion-and-reflection">Discussion and Reflection</h2>
<pre><code>- Report includes a narrative summary about how successful the analysis replications were or were not - i.e., how well do the replicated results compare to those presented in the original paper?
- Report discusses and addresses any challenges encountered... missing data, unclear information about how data were processed in the original publication, etc.
- Report discusses where and possible reasons why the analysis replications might differ from the authors' original results</code></pre>
</section>
<section id="what-to-turn-in" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="what-to-turn-in">What to Turn In</h2>
<p>Within the repo, make a “data” directory in which you include any “.csv” or “.xlsx” or similar files that contain the original data for the paper as you either downloaded them as supplementary material or received them from the paper’s author.</p>
<p>Near your own results, embed some of the original figs/tables that you replicate. You can include code in one of the following formats to reference files in your “images” folder for inclusion in your document. The first alternative uses raw HTML, the second and third accomplish the same thing using {knitr} in an {r} code block in either a <strong>Quarto</strong> (Alternative 2) document.</p>
<blockquote class="blockquote">
<p><strong>TL/DR:</strong> I should be able to CLONE your repository and knit your “.Rmd” or “.qmd” file to show all of your completed work for the data analysis replication assignment. Practically speaking, this means that if your code reads in data from external files, such as “.xlsx” or “.csv” files, it should be general enough to work on ANY machine. Thus, you will want to have your code read any data from local files housed <em>within</em> your repository , i.e., in the “data” folder described above. Your repository should also include a PDF of the original paper with the analyses being replicated at the top level of the repo.</p>
</blockquote>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>